src/overlays/actors/ovl_Arrow_Fire/z_arrow_fire.c

```diff
@@ -78,17 +80,18 @@ void ArrowFire_Charge(ArrowFire* this, GlobalContext* globalCtx) {
 
     // if arrow has no parent, player has fired the arrow
     if (arrow->actor.parent == NULL) {
-        this->unkPos = this->actor.world.pos;
+        this->trailEnd = this->actor.world.pos;
         this->radius = 10;
         ArrowFire_SetupAction(this, ArrowFire_Fly);
         this->alpha = 255;
     }
 }
 
-void func_80865ECC(Vec3f* unkPos, Vec3f* firePos, f32 scale) {
-    unkPos->x += ((firePos->x - unkPos->x) * scale);
-    unkPos->y += ((firePos->y - unkPos->y) * scale);
-    unkPos->z += ((firePos->z - unkPos->z) * scale);
+void shrinkFireTrail(Vec3f* trailEnd, Vec3f* currentPos, f32 scale) {
```

Comment:
The way functions are named is typically `SystemName_FunctionName`

Here the "system name" is `ArrowFire` (this name comes from the strings inside the debug rom, the overlay file was named "ovl_arrow_fire" originally)
For example `ArrowFire_Hit` below